### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Niv200**
Requesting merge.

### CODE REVIEW

File: client/src/final.test.js [hunk 1]
```diff
@@ -1,126 +1,139 @@
...
 let browser;
 let encoder;
 
 jest.setTimeout(30000);
-const projectName = '1.Ticket Manager UI';
+const projectName = "1.Ticket Manager UI";
 describe(projectName, () => {
   beforeAll(async () => {
-    browser = await puppeteer.launch();
+    browser = await puppeteer.launch({ headless: false, slowMo: 150 });
```
Comment (Cyber4sPopo) [inline client/src/final.test.js:45]: Remove this, so client test will pass too.

File: app.js [hunk 2]
```diff
@@ -1,6 +1,71 @@
...
+const Ticket = mongoose.model("Ticket", ticketSchema);
+
+app.get("/", async (req, res) => {
+  try {
+    res.json(arr);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({ message: "error ocurred: ", error: error.message });
+  }
+});
```
Comment (Cyber4sPopo) [inline app.js:29]: ❓

File: app.js [hunk 3]
```diff
@@ -1,6 +1,71 @@
...
+app.patch("/api/tickets/:ticketId/undone", async (req, res) => {
+  let { ticketId } = req.params;
+  try {
+    let test = await Ticket.findOneAndUpdate(ticketId, { done: false });
+    return res.json({ updated: true });
+  } catch (error) {
+    console.log(error);
+    return res.send(error);
+  }
+});
```
Comment (Cyber4sPopo) [inline app.js:51]: Everything is done good, but missing `404` handling.
Also, in `catch` you should attach appropriate status to your response (probably `200`)

File: app.js [hunk 4]
```diff
@@ -1,6 +1,71 @@
...
+
+app.get("/api/tickets", async (req, res) => {
+  let searchText = req.query.searchText;
+  try {
+    Ticket.find({}).then((result) => {
+      let arr = result;
+      if (searchText) {
+        arr = result.filter((ticket) =>
+          ticket.title.toLowerCase().includes(searchText.toLowerCase())
+        );
```
Comment (Cyber4sPopo) [inline app.js:61]: We expected filtering to be done in database level not `JS`

File: app.js [hunk 5]
```diff
@@ -1,6 +1,71 @@
...
+      }
+      res.json(arr);
+    });
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({ message: "error ocurred: ", error: error.message });
+  }
+});
+
```
Comment (Cyber4sPopo) [inline app.js:70]: Great job and great progress 👏
Pay attention to the notes to get better

**Comment by Niv200** [comment]
s

**Pull request merged**